radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java

```diff
@@ -184,19 +212,59 @@ public void process(
     }
   }
 
+  private void persistCurrentSubstateAccumulatorHash(Transaction dbTxn) {
+    var key = new DatabaseEntry(SUBSTATE_ACCUMULATOR_HASH_KEY);
+    this.substateAccumulatorHashDatabase.put(
+        dbTxn, key, new DatabaseEntry(this.currentSubstateAccumulatorHash));
+  }
+
+  private void logEpochHash() {
+    logger.info(
+        "Epoch Hash: {} for epoch {}. Time spent since last epoch: {} s.",
+        Bytes.toHexString(this.currentSubstateAccumulatorHash),
+        lastEpochInDbOpt,
+        watch.elapsed().toSeconds());
```

Comment:
If the node turns on mid transaction, this watch isn't so accurate. But maybe that's okay? What's the intended use case for this watch?

Reply:
What do you mean by mid transaction? I believe the sentence "Time spent since last epoch" won't be accurate when the node starts in the middle of an Epoch, but I think it is not a big issue. The intent is to measure the time execution overhead introduced by this feature.

Reply:
Ahh, so it just tracks the time spent doing this substate stuff? Cool - in which case, perhaps it should be renamed to something like (eg) `timeSpentOnSubstateAccumulatorThisEpoch` instead of `watch`?

Reply:
Agree, gonna change it.